Argument of type 'NexusObjectTypeDef<"Node">' is not assignable to parameter of type 'NexusInterfaceTypeDef<string>'.

Following the getting started example from the docs almost exactly:

![image](https://user-images.githubusercontent.com/5368008/52545544-e48b3100-2d7d-11e9-815f-1e1633ad5782.png)

I am unable to reference the `Node` type in my account type. 

![image](https://user-images.githubusercontent.com/5368008/52545593-3d5ac980-2d7e-11e9-92c8-a4b17ec063ee.png)

Gives me the tslint error of 

> Argument of type 'NexusObjectTypeDef<"Node">' is not assignable to parameter of type 'NexusInterfaceTypeDef<string>'.
>   Property 'config' is protected but type 'NexusObjectTypeDef<TypeName>' is not a class derived from 'NexusInterfaceTypeDef<TypeName>'.ts(2345)

And a console error of

> Using ts-node version 8.0.2, typescript version 3.3.3
> Error: Expected [object Object] to be an interfaceType, saw GraphQLObjectType
>     at SchemaBuilder.getInterface (/Users/jakelowen/code/tmp_code/nexus_test/node_modules/nexus/dist/builder.js:445:19)
>     at /Users/jakelowen/code/tmp_code/nexus_test/node_modules/nexus/dist/builder.js:169:89
>     at Array.map (<anonymous>)
>     at interfaces (/Users/jakelowen/code/tmp_code/nexus_test/node_modules/nexus/dist/builder.js:169:57)
>     at resolveThunk (/Users/jakelowen/code/tmp_code/nexus_test/node_modules/graphql/type/definition.js:374:40)
>     at defineInterfaces (/Users/jakelowen/code/tmp_code/nexus_test/node_modules/graphql/type/definition.js:522:20)
>     at GraphQLObjectType.getInterfaces (/Users/jakelowen/code/tmp_code/nexus_test/node_modules/graphql/type/definition.js:503:31)
>     at typeMapReducer (/Users/jakelowen/code/tmp_code/nexus_test/node_modules/graphql/type/schema.js:259:23)
>     at Array.reduce (<anonymous>)
>     at new GraphQLSchema (/Users/jakelowen/code/tmp_code/nexus_test/node_modules/graphql/type/schema.js:108:28)
> [ERROR] 21:47:25 Error: Expected [object Object] to be an interfaceType, saw GraphQLObjectType

I can get everything to work just fine if I do not attempt to implement the Node type and instead just specify the 'id' field manually.

Any idea why I am not able to implement Node as illustrated in the docs?

Here are my package versions:
```
"graphql": "^14.1.1",
    "graphql-yoga": "^1.17.4",
    "knex": "^0.16.3",
    "lodash": "^4.17.11",
    "nexus": "^0.9.9",
    "pg": "^7.8.0"
```

Thank you. Seems like a very promising project!

--- a/docs/getting-started.md
+++ b/docs/getting-started.md
@@ -47,12 +47,21 @@ yarn add graphql # required as a peer dependency
 As documented in the [API reference](api-core-concepts.md) GraphQL Nexus provides a consistent, scalable approach to defining GraphQL types in code.
 
 ```js
-import { objectType, queryType, stringArg, fieldArg, makeSchema } from "nexus";
+import {
+  objectType,
+  interfaceType,
+  queryType,
+  stringArg,
+  intArg,
+  fieldArg,
+  makeSchema,
+} from "nexus";
 
-const Node = objectType({
+const Node = interfaceType({
   name: "Node",
   definition(t) {
     t.id("id", { description: "Unique identifier for the resource" });
+    t.resolveType(() => null);
   },
 });
 

--- a/examples/ghost/src/schema/Query.ts
+++ b/examples/ghost/src/schema/Query.ts
@@ -1,4 +1,4 @@
-import { queryType, idArg, core } from "nexus";
+import { queryType, idArg } from "nexus";
 import { Post } from "./Post";
 import { User } from "./User";
 

--- a/src/typegenTypeHelpers.ts
+++ b/src/typegenTypeHelpers.ts
@@ -21,15 +21,17 @@ export type MaybePromise<T> = PromiseLike<T> | T;
  * resolves promises at any level of the tree, we use this
  * to help signify that.
  */
-export type MaybePromiseDeep<T> = T extends object
+export type MaybePromiseDeep<T> = Date extends T
+  ? MaybePromise<Date>
+  : boolean extends T
+  ? MaybePromise<T>
+  : T extends object
   ? MaybePromise<
       {
         [P in keyof T]: T[P] extends Array<infer U>
           ? Array<MaybePromiseDeep<U>>
           : T[P] extends ReadonlyArray<infer Y>
           ? ReadonlyArray<MaybePromiseDeep<Y>>
-          : Date extends T[P]
-          ? MaybePromise<T[P]>
           : MaybePromiseDeep<T[P]>
       }
     >

